elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -530,8 +536,16 @@ export class PeerPool extends EventEmitter {
 		return peers;
 	}
 
-	public getUniqueOutboundConnectedPeers(): ReadonlyArray<P2PPeerInfo> {
-		return getUniquePeersbyIp(this.getAllConnectedPeerInfos(OutboundPeer));
+	public getIsConnectedPeerByIP(
```

Comment:
I would call it `getConnectedPeersByFilter` and as argument, a new type with the accepted filters must be created. For now, there should be two filters by kind or by IP. Therefore, we don't need to pass the whole P2PPeerInfo object, just the IP string.